### PR TITLE
Alg/evsrestapi 387 java upgrade

### PR DESCRIPTION
**Pull request by gaaliciA1990**
Upgraded our system to use java 17, gradle 8.2.1, spring boot 2.7.17. 

Upgraded spring data elastic search to 4.2.12. 

Removed circular dependencies within the code base, resulting in a restructuring of the code, specifically in TerminologyUtils. 
- Removed the boolean `indexed` param and replaced by passing the `ElasticQueryService` or `SparqlQueryService` objects instead. 
- Added a getIndexedTerminologies. 
- Moved logic for finding a Terminology to it's own class `findTerminology` 
- Updated all calls within the codebase to use the new classes and pass the correct params

Updated the caching code to be in it's own class `SparqlQueryCacheService`. Comments explain logic

Updated `Concept` class @Fields to index only necessary fields and skip others. 

Updated tests.

### CODE REVIEW

File: .DS_Store
Comment (peter-va) [inline .DS_Store:1]: This should probably be in the gitignore since it's an OS thing for Mac

Comment (gaaliciA1990) [reply .DS_Store:1]: Thanks for catching that! I've removed the file.

File: src/test/java/gov/nih/nci/evs/api/controller/HgncSampleTest.java [hunk 1]
```diff
@@ -69,18 +62,26 @@ public void testHGNCTerminology() throws Exception {
...
+            .andExpect(status().isOk())
+            .andReturn();
     content = result.getResponse().getContentAsString();
     log.info(" content = " + content);
 
     final List<Terminology> terminologies =
-        new ObjectMapper().readValue(content, new TypeReference<List<Terminology>>() {
-          // n/a
-        });
+        new ObjectMapper()
```
Comment (peter-va) [inline src/test/java/gov/nih/nci/evs/api/controller/HgncSampleTest.java:74]: The line breaks seem rather aggressive, to the point of affecting readability. Did we change our formatting recently?

File: src/test/java/gov/nih/nci/evs/api/controller/MdrControllerTests.java [hunk 2]
```diff
@@ -260,35 +299,70 @@ public void testMdrSearchContains() throws Exception {
     ConceptResultList list = null;
 
     // Valid test
-    log.info("Testing url - " + url
-        + "?terminology=mdr&fromRecord=0&include=synonyms&pageSize=10&term=Coagulation&type=contains");
-    result = mvc.perform(get(url).header("X-EVSRESTAPI-License-Key", appProperties.getUiLicense())
-        .param("terminology", "mdr").param("term", "Coagulation").param("pageSize", "10").param("type", "contains")
-        .param("include", "synonyms")).andExpect(status().isOk()).andReturn();
+    log.info(
```
Comment (peter-va) [inline src/test/java/gov/nih/nci/evs/api/controller/MdrControllerTests.java:302]: Another example of very aggressive line breaks compromising readability. It won't affect my approving the PR, but I thought it was worth noting.

**Review by peter-va — APPROVED** [review]
Everything looks good to me. I'm a little unsure on some of the spacing changes but ultimately that's me being kinda nitpicky.

File: src/test/java/gov/nih/nci/evs/api/controller/HgncSampleTest.java (hunk 1 above)
Comment (gaaliciA1990) [reply src/test/java/gov/nih/nci/evs/api/controller/HgncSampleTest.java:74]: This was formatted using the google-java-format plugin.

**Comment by gaaliciA1990** [comment]
forced push to rebase dev onto my branch

**Comment by gaaliciA1990** [comment]
Added github action to build the project using the `make build` command.
This will not run tests since there is a dependency on elastic search and stardog, but this will make sure the code compiles and run the spotless google java formatting check to ensure we are all formatting our code consistently. If the build fails, this is also an indicator tests were not checked prior to pushing the changes.

**Comment by gaaliciA1990** [comment]
Closed to submit a new one with the incorporated changes for initializing index

**Pull request closed without merging**